src/users/users.controller.ts

```diff
@@ -36,3 +37,45 @@ export default (usersService: UsersService) => new Elysia({prefix: '/users'})
     .get("/", () => {
         return usersService.findAll();
     })
+
+
+// Idea 4: use a class that will be injected into the app state
+// and it creates its own service (or the service can be injected 
+// into the class in the same plugin that creates the class itself)
+export class UsersController {
+    private readonly usersService: UsersService;
+    private readonly db: PostgresJsDatabase
+
+    constructor(db: PostgresJsDatabase) {
+        this.db = db;
```

Comment:
Controller shouldn't be aware of the `db`, that is not its responsibility. It should only do I/O and leave the business logic to the service.